pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/TNCudaBase.hpp

```diff
@@ -196,7 +331,7 @@ class TNCudaBase : public TensornetBase<Precision, Derived> {
 
         // TODO we assign half (magic number is) of free memory size to the
         // maximum memory usage.
-        const std::size_t scratchSize = cuUtil::getFreeMemorySize() / 2;
+        std::size_t scratchSize = cuUtil::getFreeMemorySize() / 2;
```

Comment:
it is an immutable value.

```suggestion
        const std::size_t scratchSize = cuUtil::getFreeMemorySize() / 2;
```